uat/custom-components/client-ipc/src/main/java/com/aws/greengrass/testing/ipc/LocalIpcSubscriber.java

```diff
@@ -0,0 +1,166 @@
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import javax.inject.Inject;
+
+import static software.amazon.awssdk.aws.greengrass.GreengrassCoreIPCClientV2.StreamingResponse;
+
+@Log4j2
+public class LocalIpcSubscriber {
+
+    private static final int DEFAULT_PORT = 8033;
+    private static final String TOPICS_SEPARATOR = ",";
+
+    private static final String DEFAULT_CONTEXT = "ReceivedPubsubMessage";
+
+    private String assertionServerUrl = "http://localhost:8080";
+
+    @Inject
+    public LocalIpcSubscriber() {
+    }
+
+    /**
+     * Start LocalIpcSubscriber execution.
+     *
+     * @param args command line args
+     * @throws Exception on errors
+     */
+    @SuppressWarnings("PMD.SignatureDeclareThrowsException")
+    public void accept(String... args) throws Exception {
+        log.info("Args {}", args);
+
+        final String url = Optional.ofNullable(args[1])
+                                   .orElse(assertionServerUrl);
+        log.info("Assertion server URL {}", url);
+        assertionServerUrl = url;
+
+        List<String> topics = Arrays.asList(args[0].split(TOPICS_SEPARATOR));
+        subscribe(topics);
+    }
+
+    @SuppressWarnings({"PMD.AvoidCatchingGenericException", "PMD.SignatureDeclareThrowsException"})
+    private void subscribe(final List<String> topics) throws Exception {
+        log.info("Subscribe to topics {}", topics);
+
+        List<SubscribeToTopicResponseHandler> responseHandlers = new ArrayList<>();
+
+        String topic = null;
+        try (GreengrassCoreIPCClientV2 ipcClient = GreengrassCoreIPCClientV2.builder()
+                                                                            .withPort(DEFAULT_PORT)
+                                                                            .build()) {
+            for (String t : topics) {
+                topic = t;
+                SubscribeToTopicRequest request = new SubscribeToTopicRequest().withTopic(topic);
+
+                StreamingResponse<SubscribeToTopicResponse, SubscribeToTopicResponseHandler> response = 
+                            ipcClient.subscribeToTopic(request,
+                                                        this::onStreamEvent,
+                                                        Optional.of(this::onStreamError),
+                                                        Optional.of(this::onStreamClosed));
+                SubscribeToTopicResponseHandler responseHandler = response.getHandler();
+                responseHandlers.add(responseHandler);
+                log.info("Successfully subscribed to topic {}", topic);
+            }
+
+            try {
+                while (true) {
+                    Thread.sleep(10_000);
+                }
+            } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

**Remember**: Do this only for InterruptedException. Doing this for other types of exceptions may give rise to subtle bugs.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)